File: src/1109-Corporate-Flight-Bookings/1109.go
func corpFlightBookings(bookings [][]int, n int) []int {
    res := make([]int, n + 1)
    for _, book := range bookings {
        res[book[0]-1] += book[2]
        res[book[1]] -= book[2]
    }
    for i := 1; i <= n; i++ {
        res[i] += res[i-1]    
    }
    return res[:n]
}

File: src/1109-Corporate-Flight-Bookings/1109.cpp
class Solution 
{
public:
    vector<int> corpFlightBookings(vector<vector<int>>& bookings, int n) 
    {
        vector<int> res(n + 1, 0);
        for (auto& book : bookings)
        {
            res[book[0]-1] += book[2];
            res[book[1]] -= book[2];
        }
        for (int i = 1; i <= n; ++i)
        {
            res[i] += res[i-1];
        }
        return {res.begin(), res.end()-1};
    }
};